File: Js_solution/medium/Solution2487.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var removeNodes = function(head) {
    //use stack to solve this problem
    var stack = []
    var cur = head
    while (cur!=null){
        while((stack.length!==0)&&stack[stack.length-1].val<cur.val){
            stack.pop()
        }
        stack.push(cur)
        cur = cur.next;
    }
    var nxt = null;
    while(stack.length!==0){
        cur =  stack.pop();
        cur.next = nxt;
        nxt = cur;
    }
    return cur;

};

File: python_solution/medium/Solution2487.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
from typing import Optional

from medium import ListNode


class Solution:
    def removeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
         # here we use stack to solve this question
         stack = []
         cur = head
         while cur!=None:
             while stack and stack[-1].val<cur.val:
                 stack.pop()
             stack.append(cur)
             cur = cur.next
         nxt = None
         while stack:
             cur = stack.pop()
             cur.next = nxt
             nxt = cur
         return cur
